src/instance-group-manager.js

```diff
@@ -0,0 +1,574 @@
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {Operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * const Compute = require('@google-cloud/compute');
+ * const compute = new Compute();
+ * const zone = compute.zone('us-central1-a');
+ * const instanceGroupManager = zone.instanceGroupManager('web-servers');
+ *
+ * instanceGroupManager.removeVMs(function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * instanceGroupManager.removeVMs().then(function(data) {
+ *   const operation = data[0];
+ *   const apiResponse = data[1];
+ * });
+ */
+InstanceGroupManager.prototype.removeVMs = function(callback) {
+  var self = this;
+
+  this.request(
+    {
+      method: 'POST',
+      uri: '/abandonInstances',
+    },
+    function(err, resp) {
+      if (err) {
+        callback(err, null, resp);
+        return;
+      }
+
+      var operation = self.zone.operation(resp.name);
+      operation.metadata = resp;
+
+      callback(err, operation, resp);
+    }
+  );
+};
+
+/**
+ * Recreate the VMs managed by this instance group.
+ *
+ * @see [InstanceGroupManagers: recreateInstances API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers/recreateInstances}
+ *
+ * @param {VM[]} instances - Instances to recreate
+ * @param {string[]} instances - URLs to instances to recreate
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {Operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * const Compute = require('@google-cloud/compute');
+ * const compute = new Compute();
+ * const zone = compute.zone('us-central1-a');
+ * const instanceGroupManager = zone.instanceGroupManager('web-servers');
+ * const instances = [zone.vm('vm-name')];
+ *
+ * instanceGroupManager.recreateVMs(instances, function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * instanceGroupManager.recreateVMs(instances).then(function(data) {
+ *   const operation = data[0];
+ *   const apiResponse = data[1];
+ * });
+ */
+InstanceGroupManager.prototype.recreateVMs = function(instances, callback) {
+  var self = this;
+
+  var body = {
+    instances: [],
+  };
+
+  var idx, instance;
```

Comment:
Thank you for making this change. I think we can save a bit of code with something like this:

```js
var body = {
  instances: instances.map(instance => {
    if (common.until.isCustomType(instance, 'VM')) {
      return `zones/${self.zone.name}/instances/${instance.name}`;
    }
    return instance;
  }),
};
```